### PR TITLE
reduce loop count

### PR DESCRIPTION
**Pull request by giiiiiithub**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

reduce loop count

#### Which issue(s) this PR fixes:

NONE

#### Does this PR introduce a user-facing change?
NONE

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

NONE

### CODE REVIEW

File: pkg/proxy/topology.go [hunk 1]
```diff
@@ -93,6 +93,10 @@ func CategorizeEndpoints(endpoints []Endpoint, svcInfo ServicePort, nodeLabels m
 				hasLocalServingTerminatingEndpoints = true
 			}
 		}
+		
+		if hasLocalReadyEndpoints && hasLocalServingTerminatingEndpoints {
```
Comment (HirazawaUi) [inline pkg/proxy/topology.go:97]: Based on https://github.com/kubernetes/kubernetes/pull/119532#issuecomment-1660446767,  i think the change still makes sense, it does not affect the result of hasLocalServingTerminatingEndpoints, even though the improvement it brings may be marginal.
WDYT @danwinship

Comment (aojea) [reply pkg/proxy/topology.go:97]: great catch @HirazawaUi

**Comment by danwinship** [comment]
yeah, I still feel like this is doing the obviously-noticeable but fairly trivial optimization, in a way that makes the code more fragile, while ignoring the much larger potential optimization.

**Pull request closed without merging**